src/SaveAndLoad.java

```diff
@@ -0,0 +1,104 @@
+import java.io.*;
+import java.util.*;
+
+public class SaveAndLoad {
+    /**
+     * this method will return the players saved Symbols
+     * 
+     * @param fileName : the file name and directory of the Save File
+     * @return : an array of the players Symbols
+     */
+    public static char[] loadPlayerSymbol(String fileName) {
+        char player1 = 0, player2 = 0;
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fileName));
+            // Read the game state from the file
+            String line = reader.readLine();
+            player1 = line.charAt(0);
+            line = reader.readLine();
+            player2 = line.charAt(0);
+
+            reader.close();
+        } catch (IOException e) {
+            System.out.println("Error loading game: " + e.getMessage());
+        }
+        char[] playersSymbols = { player1, player2 };
+        return playersSymbols;
+    }
+
+    /**
+     * this Method will save the current Game state
+     * 
+     * @param fileName      : the directory of the save file
+     * @param player1       : player 1 Symbol
+     * @param player2       : Player 2 Symbol
+     * @param currentPlayer : the next player turn
+     * @param board         : the current status of the game board
+     * @param turn          : the number of played turns
+     */
+    public static void saveGame(String fileName, char player1, char player2, int currentPlayer, char[][] board,
+            int turn) {
+        try {
+            File saveFile = new File(fileName);
+            if (saveFile.createNewFile()) {
+                System.out.println("SaveFile created successfully");
+            }
+
+            BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
+            // Write the game state to the file
+            writer.write(String.valueOf(player1) + "\n");
+            writer.write(String.valueOf(player2) + "\n");
+            writer.write(currentPlayer + "\n");
+            writer.write(turn + "\n");
+            writer.write("Board:\n");
+            for (int i = 0; i < board.length; i++) {
```

Comment:
good job but instead of having i and j try to use meaningful name for example row and col